core/src/main/java/com/amplifyframework/api/events/ApiEndpointStatusChangeEvent.java

```diff
@@ -0,0 +1,144 @@
+    private final ApiEndpointStatus previousStatus;
+
+    /**
+     * Constructs a new {@link ApiEndpointStatusChangeEvent} object.
+     * @param currentStatus The current status of the endpoint.
+     * @param previousStatus the previous status of the endpoint.
+     */
+    public ApiEndpointStatusChangeEvent(ApiEndpointStatus currentStatus, ApiEndpointStatus previousStatus) {
+        this.currentStatus = currentStatus;
+        this.previousStatus = previousStatus;
+    }
+
+    /**
+     * Getter that returns the value of the {@link ApiEndpointStatusChangeEvent#currentStatus} field.
+     * @return The value of {@link ApiEndpointStatusChangeEvent#currentStatus} field.
+     */
+    public ApiEndpointStatus getCurrentStatus() {
+        return currentStatus;
+    }
+
+    /**
+     * Getter that returns the value of the {@link ApiEndpointStatusChangeEvent#previousStatus} field.
+     * @return The value of {@link ApiEndpointStatusChangeEvent#previousStatus} field.
+     */
+    public ApiEndpointStatus getPreviousStatus() {
+        return previousStatus;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = currentStatus != null ? currentStatus.hashCode() : 0;
+        result = 31 * result + (previousStatus != null ? previousStatus.hashCode() : 0);
+        return result;
+    }
+
+    @Override
+    public boolean equals(@Nullable Object thatObject) {
+        if (this == thatObject) {
+            return true;
+        }
+        if (thatObject == null || getClass() != thatObject.getClass()) {
+            return false;
+        }
+
+        ApiEndpointStatusChangeEvent that = (ApiEndpointStatusChangeEvent) thatObject;
+
+        if (!ObjectsCompat.equals(currentStatus, that.currentStatus)) {
+            return false;
+        }
+        return ObjectsCompat.equals(previousStatus, that.previousStatus);
+    }
+
+    @NonNull
+    @Override
+    public String toString() {
+        return "ApiEndpointStatusChangeEvent{" +
+            "currentStatus=" + currentStatus +
+            ", previousStatus=" + previousStatus +
+            "}";
+    }
+
+    @Override
+    public HubEvent<ApiEndpointStatusChangeEvent> toHubEvent() {
+        return HubEvent.create(ApiChannelEventName.API_ENDPOINT_STATUS_CHANGED, this);
+    }
+
+    /**
+     * Factory method that attempts to cast the data field of the
+     * {@link HubEvent} object as an instance of {@link ApiEndpointStatusChangeEvent}.
+     * @param hubEvent An instance of {@link HubEvent}
+     * @return An instance of {@link ApiEndpointStatusChangeEvent}.
+     * @throws AmplifyException If unable to cast to the target type.
+     */
+    public static ApiEndpointStatusChangeEvent from(HubEvent<?> hubEvent) throws AmplifyException {
+        if (hubEvent.getData() instanceof ApiEndpointStatusChangeEvent) {
+            return (ApiEndpointStatusChangeEvent) hubEvent.getData();
+        }
+        String expectedClassName = ApiEndpointStatusChangeEvent.class.getName();
```

Comment:
To keep the noise down for the reader, maybe:
```suggestion
        String expectedClassName = ApiEndpointStatusChangeEvent.class.getSimpleName();
```
?